tools/ci/pr_preview.py

```diff
@@ -121,20 +130,22 @@ def get_pull_requests(self, updated_since):
         if data['incomplete_results']:
             raise Exception('Incomplete results')
 
-        return data['items']
+        return {pr['number']: pr for pr in data['items']}
 
     @guard('core')
     def pull_request_is_from_fork(self, pull_request):
         pr_number = pull_request['number']
-        url = '{}/repos/{}/pulls/{}'.format(
-            self._host, self._github_project, pr_number
-        )
-
         logger.info('Checking if pull request %s is from a fork', pr_number)
 
-        data = gh_request('GET', url)
+        # We may already have this information; if so no need to make another
+        # API call.
+        if 'head' not in pull_request:
+            url = '{}/repos/{}/pulls/{}'.format(
+                self._host, self._github_project, pr_number
+            )
+            pull_request = gh_request('GET', url)
```

Comment:
Can we call `get_pull_request` directly?

```suggestion
            pull_request = self.get_pull_request(pr_number)
```